packages/medusa/src/strategies/batch-jobs/product/import.ts

```diff
@@ -639,11 +687,22 @@ class ProductImportStrategy extends AbstractBatchJobStrategy {
       return
     }
 
-    await this.batchJobService_.update(batchJobId, {
-      result: {
-        advancement_count: newCount,
-      },
-    })
+    await this.batchJobService_
+      .withTransaction(this.transactionManager_ ?? this.manager_)
+      .update(batchJobId, {
+        result: {
+          advancement_count: newCount,
+        },
+      })
+  }
+
+  private static buildFilename(
```

Comment:
thought(non-blocking): Would it make sense to extract this to a util as the functionality can be converted to more generic logic in a quite straightforward manner

Reply:
`withExt` could be the actual extension instead of a flag

Reply:
it si very link to the import naming. for a global util we should come up with another approach no?